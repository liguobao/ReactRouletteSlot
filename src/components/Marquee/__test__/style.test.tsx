/*
 * @Author: wzi
 * @Date: 2018-04-27 15:13:11
 * @Last Modified by: wzi
 * @Last Modified time: 2018-04-27 22:15:33
 */

import {
    Notice,
    Label,
    Horn,
    MalarqueeWrapper,
    MarqueeWrapper,
} from '../style';
import React from 'react';
import { render, shallow } from 'enzyme';
import 'jest-styled-components';
import toJson from 'enzyme-to-json';
describe('Malarquee style', () => {
    test('Notice', () => {
        const wrapper = render(<Notice />);
        expect(wrapper).toMatchSnapshot();
    });
    test('Label', () => {
        const wrapper = shallow(<Label type="pc" />);
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper).toHaveStyleRule('width', '8%');

        wrapper.setProps({ type: 'mobile' });
        expect(wrapper).toHaveStyleRule('width', '17%');
    });
    test('Horn', () => {
        const wrapper = render(<Horn type="pc" />);
        expect(wrapper).toMatchSnapshot();
    });
    test('MalarqueeWrapper', () => {
        const wrapper = shallow(<MalarqueeWrapper type="pc" />);
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper).toHaveStyleRule('width', '90%');

        wrapper.setProps({ type: 'mobile' });
        expect(wrapper).toHaveStyleRule('width', '80%');
    });
    test('MarqueeWrapper', () => {
        const wrapper = render(<MarqueeWrapper />);
        expect(wrapper).toMatchSnapshot();
    });
});
