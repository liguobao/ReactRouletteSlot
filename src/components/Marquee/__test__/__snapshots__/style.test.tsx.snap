// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Malarquee style Horn 1`] = `
.c1 {
  width: 1.5em !important;
  height: 1.5em !important;
  margin-top: 5px;
}

.c0 {
  margin-top: 0px;
}

<svg
  aria-hidden="true"
  class="icon c0 c1"
>
  <use
    href="#icon-pc"
  />
</svg>
`;

exports[`Malarquee style Label 1`] = `
.c0 {
  text-align: center;
  width: 8%;
  background: #ed6b44;
  float: left;
  color: white;
  border-radius: 4px 0px 0px 4px;
}

<span
  className="c0"
  type="pc"
/>
`;

exports[`Malarquee style MalarqueeWrapper 1`] = `
.c0 {
  display: inline-block;
  width: 90%;
}

<span
  className="c0"
  type="pc"
/>
`;

exports[`Malarquee style MarqueeWrapper 1`] = `
.c0 {
  background: white;
  border-radius: 5px;
}

<div
  class="layout-width c0"
/>
`;

exports[`Malarquee style Notice 1`] = `
.c0 {
  vertical-align: super;
  margin-left: 1px;
}

<span
  class="c0"
/>
`;
